File: src/Entity/Communaute.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

use App\Entity\Hackathon;
use Doctrine\Common\Collections\Collection;
use App\Entity\Chat;

#[ORM\Entity]
class Communaute
{

    #[ORM\Id]
    #[ORM\Column(type: "integer")]
    private int $id;

        #[ORM\ManyToOne(targetEntity: Hackathon::class, inversedBy: "communautes")]
    #[ORM\JoinColumn(name: 'id_hackathon', referencedColumnName: 'id_hackathon', onDelete: 'CASCADE')]
    private Hackathon $id_hackathon;

    #[ORM\Column(type: "string", length: 255)]
    private string $nom;

    #[ORM\Column(type: "text")]
    private string $description;

    #[ORM\Column(type: "datetime")]
    private \DateTimeInterface $date_creation;

    #[ORM\Column(type: "boolean")]
    private bool $is_active;

    public function getId()
    {
        return $this->id;
    }

    public function setId($value)
    {
        $this->id = $value;
    }

    public function getId_hackathon()
    {
        return $this->id_hackathon;
    }

    public function setId_hackathon($value)
    {
        $this->id_hackathon = $value;
    }

    public function getNom()
    {
        return $this->nom;
    }

    public function setNom($value)
    {
        $this->nom = $value;
    }

    public function getDescription()
    {
        return $this->description;
    }

    public function setDescription($value)
    {
        $this->description = $value;
    }

    public function getDate_creation()
    {
        return $this->date_creation;
    }

    public function setDate_creation($value)
    {
        $this->date_creation = $value;
    }

    public function getIs_active()
    {
        return $this->is_active;
    }

    public function setIs_active($value)
    {
        $this->is_active = $value;
    }

    #[ORM\OneToMany(mappedBy: "communaute_id", targetEntity: Chat::class)]
    private Collection $chats;
}

File: OneDrive/Bureau/Hackify-integrationn/src/main/java/models/Communaute.java
package models;

import java.sql.Timestamp;

public class Communaute {
    private int id;
    private int idHackathon; // Foreign key to hackathon
    private String nom; // Community name
    private String description; // Community description
    private Timestamp dateCreation; // Creation timestamp
    private boolean isActive; // Whether the community is active

    // Default constructor
    public Communaute() {
    }

    // Constructor for creating a new community
    public Communaute(int idHackathon, String nom, String description) {
        this.idHackathon = idHackathon;
        this.nom = nom;
        this.description = description;
        this.dateCreation = new Timestamp(System.currentTimeMillis());
        this.isActive = true;
    }

    // Constructor for retrieving from database
    public Communaute(int id, int idHackathon, String nom, String description, Timestamp dateCreation, boolean isActive) {
        this.id = id;
        this.idHackathon = idHackathon;
        this.nom = nom;
        this.description = description;
        this.dateCreation = dateCreation;
        this.isActive = isActive;
    }

    // Getters and setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdHackathon() {
        return idHackathon;
    }

    public void setIdHackathon(int idHackathon) {
        this.idHackathon = idHackathon;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Timestamp getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(Timestamp dateCreation) {
        this.dateCreation = dateCreation;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean isActive) {
        this.isActive = isActive;
    }

    @Override
    public String toString() {
        return "Communaute{" +
                "id=" + id +
                ", idHackathon=" + idHackathon +
                ", nom='" + nom + '\'' +
                ", description='" + description + '\'' +
                ", dateCreation=" + dateCreation +
                ", isActive=" + isActive +
                '}';
    }
}